internal/round/parent_round_manager.go

```diff
@@ -0,0 +1,503 @@
+}
+
+// SubmitShardRoot accepts shard root updates during the current round
+func (prm *ParentRoundManager) SubmitShardRoot(ctx context.Context, update *models.ShardRootUpdate) error {
+	prm.roundMutex.Lock()
+	defer prm.roundMutex.Unlock()
+
+	if prm.currentRound == nil {
+		return fmt.Errorf("no active round to accept shard root update")
+	}
+
+	shardKey := update.ShardID
+	prm.currentRound.ShardUpdates[shardKey] = update
+	prm.totalShardUpdates++
+
+	prm.logger.WithContext(ctx).Debug("Shard root update accepted",
+		"roundNumber", prm.currentRound.Number.String(),
+		"shardID", update.ShardID,
+		"rootHash", update.RootHash.String(),
+		"totalShards", len(prm.currentRound.ShardUpdates))
+
+	return nil
+}
+
+// StartNewRound begins a new aggregation round (public method for BFT interface)
+func (prm *ParentRoundManager) StartNewRound(ctx context.Context, roundNumber *api.BigInt) error {
+	return prm.startNewRound(ctx, roundNumber)
+}
+
+// startNewRound is the internal implementation
+func (prm *ParentRoundManager) startNewRound(ctx context.Context, roundNumber *api.BigInt) error {
+	prm.logger.WithContext(ctx).Info("Starting new parent round",
+		"roundNumber", roundNumber.String())
+
+	prm.roundMutex.Lock()
+	defer prm.roundMutex.Unlock()
+
+	// Stop existing timer if any
+	if prm.roundTimer != nil {
+		prm.roundTimer.Stop()
+	}
+
+	var shardUpdates map[int]*models.ShardRootUpdate
+	if prm.currentRound != nil {
+		shardUpdates = prm.currentRound.ShardUpdates
+	} else {
+		shardUpdates = make(map[int]*models.ShardRootUpdate)
+	}
+
+	// Create new round
+	prm.currentRound = &ParentRound{
+		Number:       roundNumber,
+		StartTime:    time.Now(),
+		State:        RoundStateCollecting,
+		ShardUpdates: shardUpdates,                   // Reuse the same map
+		Snapshot:     prm.parentSMT.CreateSnapshot(), // Create SMT snapshot for this round
+	}
+
+	// Set timer for round processing
+	prm.roundTimer = time.AfterFunc(prm.roundDuration, func() {
+		prm.logger.WithContext(ctx).Info("Parent round timer fired",
+			"roundNumber", roundNumber.String(),
+			"elapsed", prm.roundDuration.String())
+		if err := prm.processCurrentRound(ctx); err != nil {
+			prm.logger.WithContext(ctx).Error("Failed to process parent round",
+				"roundNumber", roundNumber.String(),
+				"error", err.Error())
+		}
+	})
+
+	prm.logger.WithContext(ctx).Info("Parent round started",
+		"roundNumber", roundNumber.String(),
+		"duration", prm.roundDuration.String())
+
+	return nil
+}
+
+// processCurrentRound processes the current round and creates a block
+func (prm *ParentRoundManager) processCurrentRound(ctx context.Context) error {
+	prm.roundMutex.Lock()
+
+	if prm.currentRound == nil {
+		prm.roundMutex.Unlock()
+		return fmt.Errorf("no current round to process")
+	}
+
+	// Capture current round for processing
+	round := prm.currentRound
+	round.State = RoundStateProcessing
+
+	round.ProcessedShardUpdates = make(map[int]*models.ShardRootUpdate, len(round.ShardUpdates))
```

Comment:
since ShardUpdates is carried over from the previous round this always triggers the full tree update